File: 22.Automated-Testing-Unit-Integration-and-E2E/code/typescript/fixtures_factories_8.ts
function newUser(opts: any = {}) {
    return {
        id: "u1",
        email: "default@x.com",
        active: true,
        ...opts
    };
}

describe('newUser', () => {
    it('overrides defaults', () => {
        const u = newUser({ email: 'ada@x.com' });
        expect(u.email).toBe('ada@x.com');
    });
});

File: 22.Automated-Testing-Unit-Integration-and-E2E/code/rust/fixtures_factories_8.rs
struct User {
    id: String,
    email: String,
    active: bool,
}

fn new_user(email_override: Option<String>) -> User {
    User {
        id: "u1".into(),
        email: email_override.unwrap_or_else(|| "default@x.com".into()),
        active: true,
    }
}

#[test]
fn test_signup() {
    let u = new_user(Some("ada@x.com".into()));
    assert_eq!(u.email, "ada@x.com");
}
